src/Compilers/Core/AnalyzerDriver/AnalyzerDriverHelper.cs

```diff
@@ -0,0 +1,557 @@
+			foreach (var operationAction in operationActions)
+			{
+				foreach (var kind in operationAction.Kinds)
+				{
+					ArrayBuilder<OperationAnalyzerAction> actionsForKind;
+					if (!operationActionsByKind.TryGetValue(kind, out actionsForKind))
+					{
+						operationActionsByKind.Add(kind, actionsForKind = ArrayBuilder<OperationAnalyzerAction>.GetInstance());
+					}
+
+					actionsForKind.Add(operationAction);
+				}
+			}
+
+			var tuples = operationActionsByKind.Select(kvp => KeyValuePair.Create(kvp.Key, kvp.Value.ToImmutableAndFree()));
+			var map = ImmutableDictionary.CreateRange(tuples);
+			operationActionsByKind.Free();
+			return map;
+		}
+
+		internal static void ExecuteOperationActions(
+            IEnumerable<IOperation> operationsToAnalyze,
+            ImmutableDictionary<OperationKind, ImmutableArray<OperationAnalyzerAction>> operationActionsByKind,
+            AnalyzerOptions analyzerOptions,
+            Action<Diagnostic> addDiagnostic,
+            Func<Exception, DiagnosticAnalyzer, bool> continueOnException,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(operationActionsByKind != null);
+            Debug.Assert(operationActionsByKind.Any());
+
+            foreach (IOperation operation in operationsToAnalyze)
+            {
+                ImmutableArray<OperationAnalyzerAction> actionsForKind;
+                if (operationActionsByKind.TryGetValue(operation.Kind, out actionsForKind))
+                {
+                    foreach (OperationAnalyzerAction action in actionsForKind)
+                    {
+                        ExecuteOperationAction(action.Action, operation, action.Analyzer, analyzerOptions, addDiagnostic, continueOnException, cancellationToken);
+                    }
+                }
+            }
+        }
+
+        internal static bool CanHaveExecutableCodeBlock(ISymbol symbol)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Method:
+                case SymbolKind.Event:
+                case SymbolKind.Property:
+                case SymbolKind.NamedType:
+                    return true;
+
+                case SymbolKind.Field:
+                    Debug.Assert(((IFieldSymbol)symbol).AssociatedSymbol == null);
+                    return true;
+
+                default:
+                    return false;
+            }
+        }
+
+        internal static void ExecuteAndCatchIfThrows(DiagnosticAnalyzer analyzer, Action<Diagnostic> addDiagnostic, Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException, Action analyze, CancellationToken cancellationToken)
+        {
+            try
+            {
+                analyze();
+            }
+            catch (OperationCanceledException oce) when (continueOnAnalyzerException(oce, analyzer))
+            {
+                if (oce.CancellationToken != cancellationToken)
```

Comment:
This source file needs to be removed from the PR. (It was deleted in an earlier merged PR, and I kept it around so that I could copy changed code out of it. It won't be checked in.)